smart_campus/app/views.py

```diff
@@ -38,6 +40,16 @@
 )
 
 
+def validate_administrator(function=None):
+    @wraps(function)
+    def wrapper(request):
```

Comment:
Not all view functions take only argument `request`.
You should use `*args, **kwargs` so that it could be used elsewhere

Reply:
Okay, I will add them.